starknet_py/contract.py

```diff
@@ -185,17 +200,25 @@ async def invoke(
         if max_fee is not None:
             self.max_fee = max_fee
 
-        transaction = await self._account_client.sign_invoke_transaction(
-            calls=self,
-            max_fee=self.max_fee,
-            auto_estimate=auto_estimate,
-            version=self.version,
-        )
-        response = await self._account_client.send_transaction(transaction)
+        if isinstance(self._account, _AccountProxy):
+            transaction = await self._account.sign_invoke_transaction(
+                calls=self,
+                max_fee=self.max_fee,
+                auto_estimate=auto_estimate,
+                version=self.version,
```

Comment:
What is the purpose of this? The old account had a version provided in constructor, right?

Reply:
Yes, originally I tried adding `**kwargs` to the BaseAccount interface so it would optionally accept `version` but it was weird to have that there. I think we can do with a few `isinstance` checks like this instead.

Reply:
But why do we need to provide version here?

Reply:
For some reason `AccountClient` accepts optional `version` parameter in methods like `sign_invoke_transaction` even though it also has accepted tx version in `self`. 

We could omit it and rely on the default behavior but perhaps this is safer?

Reply:
Actually it seems that if you set `version=0` on new account using version 1, you'll get a `ValueError` so even more illogical.  I agree we can remove it from Contract.